### PR TITLE
Use Dropout1d for the channel-wise dropout in BDTCN, TCN, BENDR

### PR DESCRIPTION
**Pull request by tayal-sarthak**
Problem
`_TemporalBlock` in `braindecode/models/tcn.py` applies `nn.Dropout2d` to activations
shaped `(batch, channels, times)`. `_ConvEncoderBENDR` in `braindecode/models/bendr.py`
does the same after each `nn.Conv1d` stage. PyTorch routes 3D input to `Dropout2d`
through a deprecated compatibility path, which warns at each forward pass:

```
dropout2d: Received a 3D input to dropout2d and assuming that channel-wise 1D dropout
behavior is desired - input is interpreted as shape (N, C, L), where C is the channel dim.
This behavior will change in a future release to interpret the input as one without a batch
dimension, i.e. shape (C, H, W). To maintain the 1D channel-wise dropout behavior, please
switch to using dropout1d instead.
```

On current master a single training forward pass raises six of these warnings in
`BDTCN`, six in `BENDR`:

```python
import warnings, torch
from braindecode.models import BDTCN, BENDR

for cls in (BDTCN, BENDR):
    model = cls(n_chans=8, n_outputs=2, n_times=256, sfreq=100.0).train()
    with warnings.catch_warnings(record=True) as caught:
        warnings.simplefilter("always")
        model(torch.randn(4, 8, 256))
    print(cls.__name__, len([w for w in caught if "dropout2d" in str(w.message)]))
# BDTCN 6
# BENDR 6
```

Beyond the noise, the announced reinterpretation reads dimension 0 as the channel axis.
Masking would then drop whole batch items in place of feature channels, changing the
training behaviour of these models with nothing surfacing at runtime.

So, the change

`nn.Dropout2d` becomes `nn.Dropout1d` at the call sites receiving 3D activations:

| File | Site | Models |
| --- | --- | --- |
| `braindecode/models/tcn.py` | `_TemporalBlock.dropout1`, `_TemporalBlock.dropout2` | `BDTCN`, `TCN` |
| `braindecode/models/bendr.py` | `_ConvEncoderBENDR` encoder stages | `BENDR`, `InterpolatedBENDR` |

Call sites receiving 4D activations keep `nn.Dropout2d`, namely `ATCNet`, `TIDNet`.

#424 records the reason `Dropout2d` sits in the TCN residual block. The goal is masking
complete channels in place of single points. The BENDR encoder mirrors the reference
`dn3_ext.py` implementation, which places dropout after `nn.Conv1d` with the same
channel-masking intent. `nn.Dropout1d` expresses that semantic directly on `(N, C, L)`
input, with no deprecation attached. It is available since torch 1.12, well below the
`torch>=2.0` floor declared in `pyproject.toml`

### CODE REVIEW

**Review by adammounir — APPROVED** [review]
hey @tayal-sarthak, I just verified locally, full test_models.py passes and the dropout2d warnings are gone; LGTM thank you 👍
